metals/src/main/scala/scala/meta/internal/metals/DefinitionProvider.scala

```diff
@@ -334,7 +337,9 @@ class DestinationProvider(
   ): Option[DefinitionDestination] = {
     definition(symbol, allowedBuildTargets).map { defn =>
       val destinationDoc = bestTextDocument(defn)
-      val destinationPath = defn.path.toFileOnDisk(workspace)
+      val destinationPath =
+        if (saveSymbolFileToDisk) defn.path.toFileOnDisk(workspace)
```

Comment:
This makes sure that we don't save the file to disk, but I can understand how the file is being sent back to the user. Do we send the path with the jar and VS Code is responsible for that?

Reply:
Yes - the wonder of Virtual documents.  Metals tells VSCode (or any client) to open a jar URI e.g. `jar:file://path/jarname.jar!source/source.java` instead of the URI of the read-only file that we've extracted from the jar and saved on the file system.  VSCode intercepts any URI with the `jar` scheme and asks the `MetalsFileProvider` to provide the contents of that document.  `MetalsFileProvider` sends a `DecodeFile` command to Metals with this same URI. Metals hands this off to the `FileDecoderProvider` which extracts the file's contents from the jar and sends it back to VSCode.  No persistence to the filesystem occurs.  It's quite a nice way (I think) of handling any data that we don't have saved on the file system but refer to by URI.  Then you can use standard jar style URIs to navigate round the source jars or still create read-only files and still navigate round source jars using `file` scheme URIs.  If all clients supported this then Metals could dump the readonly folder entirely (I think)

Reply:
Thanks for explanation!